components/binding-usage-controller/internal/controller/labels_fetcher.go

```diff
@@ -38,35 +40,64 @@ func (c *BindingLabelsFetcher) Fetch(svcBinding *scTypes.ServiceBinding) (map[st
 		return nil, errors.Wrapf(err, "while fetching ServiceInstance [%s] from namespace [%s] indicated by ServiceBinding", svcInstanceName, svcBinding.Namespace)
 	}
 
-	svcClassName := c.getClassNameFromInstance(svcInstance)
-	if svcClassName == "" {
-		return nil, fmt.Errorf("cannot fetch ClusterServiceClass from [%s]", pretty.ServiceInstanceName(svcInstance))
-	}
+	svcClusterServiceClassName := c.getClusterServiceClassNameFromInstance(svcInstance)
+	svcServiceClassName := c.getServiceClassNameFromInstance(svcInstance)
 
-	svcClass, err := c.scLister.Get(svcClassName)
-	if err != nil {
-		return nil, errors.Wrapf(err, "while fetching ClusterServiceClass [%s]", svcClassName)
-	}
+	switch {
+	case svcClusterServiceClassName != "":
+		svcClass, err := c.cscLister.Get(svcClusterServiceClassName)
+
```

Comment:
**minor of minor:** you can remove the new line, so it becomes one block, same as you already have below for labels, this groups steps nicely 👍  

the same suggestion in _line 62_

Reply:
done